File: src/main/java/com/intellij/tapestry/core/model/externalizable/documentation/generationchain/ServiceDocumentation.java
package com.intellij.tapestry.core.model.externalizable.documentation.generationchain;

import com.intellij.tapestry.core.model.externalizable.documentation.Home;
import com.intellij.tapestry.core.util.VelocityProcessor;

import java.util.Map;

/**
 * Renders the documentation page for a single Tapestry IoC service.
 */
public final class ServiceDocumentation {

    private static final String TEMPLATE = "/documentation/service.vm";

    private ServiceDocumentation() {
    }

    public static String render(Home.ServiceDoc service) {
        Map<String, Object> context = AbstractDocumentationGenerator.baseContext();
        context.put("service", service);
        return VelocityProcessor.processClasspathTemplate(TEMPLATE, context);
    }
}

File: src/main/kotlin/com/intellij/tapestry/core/model/externalizable/documentation/generationchain/ServiceDocumentation.kt
package com.intellij.tapestry.core.model.externalizable.documentation.generationchain

import com.intellij.tapestry.core.model.externalizable.documentation.Home
import com.intellij.tapestry.core.util.VelocityProcessor

/**
 * Renders the documentation page for a single Tapestry IoC service.
 */
object ServiceDocumentation {

    private const val TEMPLATE = "/documentation/service.vm"

    fun render(service: Home.ServiceDoc): String {
        val context = DocAssets.baseContext()
        context["service"] = service
        return VelocityProcessor.processClasspathTemplate(TEMPLATE, context)
    }
}
